### ISSUE TITLE
Test TestComputeInstanceMigrateState panics on run

### ISSUE THREAD

**Issue by selmanj** [terraform-providers/terraform-provider-google#557]
Looks like the test depends on some global state being initialized and fails when run independently. See output below.

```
$ go test -v ./google -run "TestComputeInstanceMigrateState"
=== RUN   TestComputeInstanceMigrateState
2017/10/09 11:54:24 [INFO] Found Compute Instance State v0; migrating to v1
2017/10/09 11:54:24 [DEBUG] Attributes before migration: map[string]string{"metadata.2.with.dots":"should.work", "metadata.#":"2", "metadata.0.foo":"bar", "metadata.1.baz":"qux"}
2017/10/09 11:54:24 [DEBUG] Attributes after migration: map[string]string{"metadata.with.dots":"should.work", "metadata.foo":"bar", "metadata.baz":"qux"}
2017/10/09 11:54:24 [INFO] Found Compute Instance State v1; migrating to v2
2017/10/09 11:54:24 [DEBUG] Attributes before migration: map[string]string{"metadata.baz":"qux", "metadata.with.dots":"should.work", "metadata.foo":"bar"}
2017/10/09 11:54:24 [DEBUG] Attributes after migration: map[string]string{"metadata.baz":"qux", "metadata.with.dots":"should.work", "metadata.foo":"bar"}
2017/10/09 11:54:24 [INFO] Found Compute Instance State v2; migrating to v3
2017/10/09 11:54:24 [DEBUG] Attributes before migration: map[string]string{"metadata.baz":"qux", "metadata.with.dots":"should.work", "metadata.foo":"bar"}
2017/10/09 11:54:24 [DEBUG] Attributes after migration: map[string]string{"metadata.baz":"qux", "metadata.with.dots":"should.work", "metadata.foo":"bar", "create_timeout":"4"}
2017/10/09 11:54:24 [INFO] Found Compute Instance State v3; migrating to v4
2017/10/09 11:54:24 [DEBUG] Attributes before migration: map[string]string{"metadata.baz":"qux", "metadata.with.dots":"should.work", "metadata.foo":"bar", "create_timeout":"4"}
--- FAIL: TestComputeInstanceMigrateState (0.00s)
panic: interface conversion: interface {} is nil, not *google.Config [recovered]
	panic: interface conversion: interface {} is nil, not *google.Config

goroutine 5 [running]:
testing.tRunner.func1(0xc4202e4000)
	/usr/local/go/src/testing/testing.go:711 +0x2d2
panic(0x2020ce0, 0xc420208d00)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/terraform-providers/terraform-provider-google/google.migrateStateV3toV4(0xc420087090, 0x0, 0x0, 0x0, 0x48, 0xc420087090)
	/Users/jsselman/src/go-work/src/github.com/terraform-providers/terraform-provider-google/google/resource_compute_instance_migrate.go:182 +0x1fb5
github.com/terraform-providers/terraform-provider-google/google.resourceComputeInstanceMigrateState(0x0, 0xc420087090, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jsselman/src/go-work/src/github.com/terraform-providers/terraform-provider-google/google/resource_compute_instance_migrate.go:48 +0x21a
github.com/terraform-providers/terraform-provider-google/google.runInstanceMigrateTest(0xc4202e4000, 0x22451d4, 0x8, 0x224fea9, 0x12, 0x0, 0xc4202c89c0, 0xc4202c89f0, 0x0, 0x0)
	/Users/jsselman/src/go-work/src/github.com/terraform-providers/terraform-provider-google/google/resource_compute_instance_migrate_test.go:777 +0xc1
github.com/terraform-providers/terraform-provider-google/google.TestComputeInstanceMigrateState(0xc4202e4000)
	/Users/jsselman/src/go-work/src/github.com/terraform-providers/terraform-provider-google/google/resource_compute_instance_migrate_test.go:71 +0xc84
testing.tRunner(0xc4202e4000, 0x22b9a88)
	/usr/local/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:789 +0x2de
exit status 2
FAIL	github.com/terraform-providers/terraform-provider-google/google	0.138s
```

**Comment by selmanj** [comment]
I was wrong; the test fails every time (or skips if `TF_ACC=1` is not set; likely this is why it wasn't caught as most unit tests don't have this issue).

Updating title

**Comment by paddycarver** [comment]
So I've dug into this, and this is what I've found:

* Travis runs tests on every PR. But because our acceptance tests cost money and take approximately forever (ok, like 2 hours, but still too long) Travis doesn't run them. They're skipped unless the `TF_ACC` environment variable is set.
* Our CI server runs the acceptance tests with that environment variable set every night. For reasons I'm not familiar with, it only runs tests that are named `func TestAcc*`, however.
* These tests _used_ to be run by Travis, as they didn't need to hit the API. However, once we needed to migrate the disks, we hit the API, so these had to be converted to acceptance tests. We correctly added the environment variable filtering (the test skips if `TF_ACC` isn't set, as it should) because Travis tried running it and it failed. However, we failed to update the test name to match `TestAcc`, so our CI never ran it.

And that's how we got a test that neither of our automated systems ran.

To fix this issue:

* Let's rename the test to start with `TestAcc`, so CI runs it.
* The test needs to have a `*google.Config` set as `meta`, not a nil, or it panics.
* The tests from before disks were migrated need to be updated to actually stand up an instance in the desired state, as the tests for disk migration do, so the migration function can successfully run. This is because migration tests test migration from every version of state to the _current_ version of state, so even if we don't care about the results of the disk migration, they'll still happen, and we need an instance for them to happen against.

In the future:

* Finding some way to find tests that slipped through both Travis and our CI is obviously desirable. In theory, we could add a check to Travis that ensures every skipped test begins with `TestAcc` and errors if not. Alternatively, we could just run every test in CI, not just those that begin with `TestAcc`.
* Having a better framework for testing migrations that need to actually hit the API is clearly something we can improve on.

I count 61 tests right now that do not get run by our CI server nightly. I'll check right now to make sure none of them are being skipped if `TF_ACC` is not set, which would mean they're not getting run at all.
